content/tutorials/custom-zk-chain/20.customizing-your-chain.md

```diff
@@ -0,0 +1,360 @@
+---
+title: Customizing your ZK chain
+description: Create a ZK chain with a custom ERC20 base token.
+
+---
+
+One way you can customize your ZK chain is by changing the base token used for gas from ETH to an ERC20 token.
+Let's try customizing a new chain by using a custom ERC20 base token.
+
+For the first release of the ZK stack, the base tokens you can use for your chain must be whitelisted.
+The whitelisted addresses are stored and checked in the `BridgeHub` contract on the L1.
+In the future it will be possible to add a new token in a permissionless proccess.
+
+For now, you have the ability to whitelist tokens in your local ecosystem.
+
+The overall flow for setting up a local chain with a custom ERC20 base token looks like this:
+
+1. Deploy an ERC20 token to the L1.
+1. Send ERC20 tokens to the governor address on the L1.
+1. Create your new chain that uses the ERC20 token and set it as the default.
+1. Initialize the new chain in the ecosystem.
+1. Bridge tokens from the L1 to your new chain.
+
+## Deploying an ERC20 token
+
+For the first step of this flow, let's make a new hardhat project.
+
+Move out of your previous folder and make a new folder for the ERC20 token contract.
+
+```bash
+mkdir base-token-contract
+cd base-token-contract
+```
+
+Then, run the `hardhat init` command to generate a new project:
+
+```bash
+npx hardhat init
+```
+
+Select the option **Create a Typescript project** and accept the default options for everything else.
+
+Run the command below to install the necessary dependencies:
+
+::code-group
+
+```bash [npm]
+npm i -D typescript ts-node @openzeppelin/contracts @nomicfoundation/hardhat-ethers @typechain/ethers-v6 @typechain/hardhat typechain ethers dotenv
+```
+
+```bash [yarn]
+yarn add -D typescript ts-node @openzeppelin/contracts @nomicfoundation/hardhat-ethers @typechain/ethers-v6 @typechain/hardhat typechain ethers dotenv
+```
+
+::
+
+Once installed, replace your existing config in `hardhat.config.ts` with the config below:
+
+```ts
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";
+
+import dotenv from "dotenv";
+dotenv.config();
+
+const config: HardhatUserConfig = {
+  solidity: "0.8.20",
+  networks: {
+        localRethNode: {
+          url: "http://localhost:8545",
+          accounts: [process.env.WALLET_PRIVATE_KEY as any],
+        },
+      },
+};
+
+export default config;
+```
+
+Next, create a `.env` file with:
+
+```bash
+touch .env
+```
+
+and add the `WALLET_PRIVATE_KEY` environment variable with the private key of the rich wallet we've been using:
+
+```env
+WALLET_PRIVATE_KEY=0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110
```

Comment:
I now realize this is just a representation of the address of whose private key we want to use, but it feels a little confusing seeing it this way. It's very tempting to just copy/paste and I see a lot of users doing that without paying attention to what is being represented here.

Reply:
I added a couple notes for this, but not sure it addresses your concern